lib/HTTP/Daemon.pm

```diff
@@ -38,15 +42,27 @@ sub url {
     my $self = shift;
     my $url  = $self->_default_scheme . "://";
     my $addr = $self->sockaddr;
-    if (!$addr || $addr eq INADDR_ANY) {
+    if (!$addr || $addr eq INADDR_ANY || $addr eq IN6ADDR_ANY) {
         require Sys::Hostname;
         $url .= lc Sys::Hostname::hostname();
     }
     elsif ($addr eq INADDR_LOOPBACK) {
-        $url .= inet_ntoa($addr);
+        $url .= inet_ntop(AF_INET, $addr);
+    }
+    elsif ($addr eq IN6ADDR_LOOPBACK) {
+        $url .= '[' . inet_ntop(AF_INET6, $addr) . ']';
     }
     else {
-        $url .= gethostbyaddr($addr, AF_INET) || inet_ntoa($addr);
+        my $host = $self->sockhostname;
+        if (!defined $host) {
+            if (sockaddr_family($addr) eq AF_INET6) {
```

Comment:
https://metacpan.org/pod/Socket#$family-=-sockaddr_family-$sockaddr
It seems that sockaddr_family() expects sockname, not sockaddr.
In fact, in my environment, `sockaddr_family($addr)` returns 36, which is neither AF_INET nor AF_INET6.

So these lines should be
```diff
diff --git lib/HTTP/Daemon.pm lib/HTTP/Daemon.pm
index a760c82..10b1f62 100644
--- lib/HTTP/Daemon.pm
+++ lib/HTTP/Daemon.pm
@@ -55,11 +55,15 @@ sub url {
     else {
         my $host = $self->sockhostname;
         if (!defined $host) {
-            if (sockaddr_family($addr) eq AF_INET6) {
+            my $sockname = $self->sockname;
+            if (sockaddr_family($sockname) eq AF_INET6) {
                 $host = '[' . inet_ntop(AF_INET6, $addr) . ']';
             }
+            elsif (sockaddr_family($sockname) eq AF_INET) {
+                $host = inet_ntop(AF_INET, $addr);
+            }
             else {
-                $host = inet_ntop(AF_INET6, $addr);
+                die "Unknown family";
             }
         }
         $url .= $host;
```

Reply:
Also a good catch!